CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Camera/Views/QRCodeOverlay.swift

```diff
@@ -0,0 +1,259 @@
+        return view
+    }()
+
+    private lazy var incorrectQRFeedback: IncorrectQRCodeTextContainer = {
+        let view = IncorrectQRCodeTextContainer()
+        view.layer.cornerRadius = Constants.spacing
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.isHidden = true
+        return view
+    }()
+
+    private lazy var checkMarkImageView: UIImageView = {
+        let imageView = UIImageView()
+        imageView.image = UIImageNamedPreferred(named: "greenCheckMark")
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.isHidden = true
+        return imageView
+    }()
+
+    private lazy var loadingIndicatorView: UIActivityIndicatorView = {
+        let indicatorView = UIActivityIndicatorView()
+        indicatorView.hidesWhenStopped = true
+        indicatorView.style = .whiteLarge
+        return indicatorView
+    }()
+
+    private lazy var loadingIndicatorText: UILabel = {
+        var loadingIndicatorText = UILabel()
+        loadingIndicatorText.font = configuration.textStyleFonts[.bodyBold]
+        loadingIndicatorText.textAlignment = .center
+        loadingIndicatorText.adjustsFontForContentSizeCategory = true
+        loadingIndicatorText.textColor = .GiniCapture.light1
+        loadingIndicatorText.isAccessibilityElement = true
+        loadingIndicatorText.text = NSLocalizedStringPreferredFormat("ginicapture.QRscanning.loading",
+                                                                     comment: "Retrievenig invoice")
+        return loadingIndicatorText
+    }()
+
+    private lazy var loadingContainer: UIStackView = {
+        let textStackView = UIStackView()
+        textStackView.axis = .vertical
+        textStackView.distribution = .fillProportionally
+        textStackView.spacing = Constants.spacing * 2
+        textStackView.translatesAutoresizingMaskIntoConstraints = false
+        textStackView.isHidden = true
+        return textStackView
+    }()
+
+    init() {
+        super.init(frame: .zero)
+        addSubview(correctQRFeedback)
+        addSubview(checkMarkImageView)
+        addSubview(incorrectQRFeedback)
+
+        addLoadingView()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func addLoadingView() {
+        let loadingIndicator: UIView
+
+        if let customLoadingIndicator = configuration.analysisScreenLoadingIndicator?.injectedView() {
+            loadingIndicator = customLoadingIndicator
+        } else {
+            loadingIndicator = loadingIndicatorView
+        }
+
+        addSubview(loadingContainer)
+        loadingContainer.addArrangedSubview(loadingIndicator)
+        loadingContainer.addArrangedSubview(loadingIndicatorText)
+    }
+
+    func layoutViews(centeringBy cameraFrame: UIView) {
+        NSLayoutConstraint.activate([
+            incorrectQRFeedback.topAnchor.constraint(equalTo: cameraFrame.topAnchor,
```

Comment:
[Suggestion] could you please extract into small methods the layout changes: incorrectQRFeedbackLayout(), correctQRFeedbackLayout() and etc?